internal/config/git.go

```diff
@@ -10,13 +10,17 @@ import (
 
 // Push registers Git push functionality for the given repo and key.
 func (cfg *Config) Push(repo string, pk ssh.PublicKey) {
-	err := cfg.Reload()
-	if err != nil {
-		log.Printf("error reloading after push: %s", err)
-	}
-	if cfg.Cfg.Callbacks != nil {
-		cfg.Cfg.Callbacks.Push(repo)
-	}
+	go func() {
+		cfg.reloadMtx.Lock()
```

Comment:
I'd actually put this in `Config.Reload()` since that will have the same issue if called elsewhere. It will block this call too, so effectively make it thread safe.

Reply:
Done